### PR TITLE
[MRG] Make MondrianTreeRegressor pass estimator_checks

### PR DESCRIPTION
**Pull request by MechCoder**

### CODE REVIEW

**Comment by glouppe** [comment]
Could you try to fix this?

**Comment by MechCoder** [comment]
It is not completely straightforward. For instance, see https://github.com/scikit-learn/scikit-learn/issues/8683. But I'll see what I can do.

**Comment by MechCoder** [comment]
Actually it was pretty straightforward. Merging when tests pass.

File: mondrian_forest/ensemble/forest.py [hunk 1]
```diff
@@ -58,6 +59,32 @@ def __init__(self,
...
+            ignored while searching for a split in each node. In the case of
+            classification, splits are also ignored if they would result in any
+            single class carrying a negative weight in either child node.
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        # multi_output is not supported.
+        X, y = check_X_y(X, y, dtype=np.float32, multi_output=False)
```
Comment (MechCoder) [inline mondrian_forest/ensemble/forest.py]: Removed multioutput and sparse since we do not support them in master.

**Comment by MechCoder** [comment]
It's not clear to me why `check_estimators_unfitted` fails. Atleast it passes for me locally. Do you have any intuition?

```python
 mfr = MondrianForestRegressor()
X, _ = make_regression()
mfr.predict(X)
NotFittedError: This MondrianForestRegressor instance is not fitted yet. Call 'fit' with appropriate arguments before using this method.
```

**Comment by MechCoder** [comment]
Seems like `estimators_` is set at init time. Trying to figure out why.

**Comment by glouppe** [comment]
In sklearn 0.18.X, `estimators_` was set in the constructor.
https://github.com/scikit-learn/scikit-learn/blob/0.18.X/sklearn/ensemble/base.py#L92

However, this line has been removed in sklearn#master.

**Comment by MechCoder** [comment]
Merging. Thanks @glouppe !

**Pull request merged**